flow360/component/flow360_params/flow360_params.py

```diff
@@ -73,6 +73,98 @@ class MeshBoundary(Flow360BaseModel):
     no_slip_walls: Union[List[str], List[int]] = pd.Field(alias="noSlipWalls")
 
 
+class TurbulenceValueSpecification(Flow360BaseModel):
+    """Turbulence values on boudaries or freestream
+
+    Parameters
+    ----------
+    turbulentKineticEnergy : non-dimensional [`C_inf^2`]
+        Turbulent kinetic energy. Applicable only when using SST model.
+
+    turbulentIntensity : non-dimensional [`-`]
+        Turbulent intensity. Applicable only when using SST model.
+        This is related to turbulent kinetic energy as: 
+        `turbulentKineticEnergy = 1.5*pow(U_ref * turbulentIntensity, 2)`. 
+        Note the use of the freestream velocity U_ref instead of C_inf.
+
+    specificDissipationRate : non-dimensional [`C_inf/L_gridUnit`]
+        Turbulent specific dissipation rate. Applicable only when using SST model.
+
+    turbulentViscosityRatio : non-dimensional [`-`]
+        The ratio of turbulent eddy viscosity over the freestream viscosity. Applicable for both SA and SST model.
+
+    turbulentLengthScale : non-dimensional [`L_gridUnit`]
+        The turbulent length scale is an estimation of the size of the eddies that are modeled/not resolved. 
+        Applicable only when using SST model. This is related to the turbulent kinetic energy and turbulent
+        specific dissipation rate as: `L_T = sqrt(k)/(pow(beta_0^*, 0.25)*w)` where `L_T` is turbulent length scale,
+        `k` is turbulent kinetic energy, `beta_0^*` is 0.09 and `w` is turbulent specific dissipation rate. 
+        Applicable only when using SST model.
+
+    modifiedTurbulentViscosityRatio : non-dimensional [`-`]
+        The ratio of modified turbulent eddy viscosity (SA) over the freestream viscosity. 
+        Applicable only when using SA model.
+
+    modifiedTurbulentViscosity : non-dimensional [`C_inf*L_gridUnit`]
+        The modified turbulent eddy viscosity (SA). Applicable only when using SA model.
+
+    """
+
+    turbulentKineticEnergy: Optional[NonNegativeFloat] = pd.Field(alias="turbulentKineticEnergy")
+    turbulentIntensity: Optional[NonNegativeFloat] = pd.Field(alias="turbulentIntensity")
+    specificDissipationRate: Optional[NonNegativeFloat] = pd.Field(alias="specificDissipationRate")
+    turbulentViscosityRatio: Optional[NonNegativeFloat] = pd.Field(alias="turbulentViscosityRatio")
+    turbulentLengthScale: Optional[PositiveFloat] = pd.Field(alias="turbulentLengthScale")
+    modifiedTurbulentViscosityRatio: Optional[PositiveFloat] = pd.Field(
+        alias="modifiedTurbulentViscosityRatio"
+    )
+    modifiedTurbulentViscosity: Optional[PositiveFloat] = pd.Field(
+        alias="modifiedTurbulentViscosity"
+    )
+
+    # pylint: disable=no-self-argument
+    @pd.root_validator(pre=True)
+    def validate_validate_combination(cls, value: dict):
+        """
```

Comment:
lets try to achieve the same based on data structure without using custom validations. Custom validations are problematic to export to webUI via JSON schema whereas data structure can easily be exported.

for example:
```python
     set(["specificDissipationRate", "turbulentLengthScale"]),
```
can be done using:
```python
SpecificDissipationRateAndTurbulentLengthScale(Flow360BaseModel):
    specificdissipationrate: Optional[NonNegativeFloat] = pd.Field(alias="specificDissipationRate")
    turbulentlengthscale: Optional[PositiveFloat] = pd.Field(alias="turbulentLengthScale")
```

and then:
```python
class FreestreamBoundary(Boundary):
     """Freestream boundary"""

     type = pd.Field("Freestream", const=True)
     velocity: Optional[BoundaryVelocityType] = pd.Field(alias="Velocity")
     turbulence_value_specification: Optional[Union[SpecificDissipationRateAndTurbulentLengthScale, ...] = pd.Field(alias="turbulenceValueSpecification")
```
so `Union` will be responsible for all possible combinations and validations. When you provide 3, then pydantic will not be able to find valid a type for it.

the solver side would need to change:
from `boundaries -> name -> specificDissipationRate`
to `boundaries -> name -> turbulenceValueSpecification -> specificDissipationRate`

Reply:
I am concerned that the user will need to write something like:

`bc = FreestreamBoundary(turbulenceQuantities=SpecificDissipationRateAndTurbulentLengthScale(specificdissipationrate=0.2, turbulentlengthscale = 0.5))`

 which is super long. We can move this (check valid combinations) to the root validator. We will need a root validator for this PR anyway for checking SST and SA compatible combinations.

Reply:
Agree, it is now very long. I think we can create a function for the user for easier input:
```python
FreestreamBoundary(turbulence_quantities=turbulence_quantities(dissipation_rate=0.2, length_scale=0.5))
```
the function can then try to match type using pydantic and return object of a correct type. This way we can keep the data structure validation and have reasonable input way for the user.

Reply:
Few other ideas:
1. We can remove `turbulent` from parameter names if they are provided to a function/class with turbulent in its name. 
2. The proposed function turbulence_quantities() can also named using class naming convention `TurbulenceQuantities()` if that will be more natural for the user in this specific instance

Reply:
Removed some "turbulent" as the others sounds strange without turbulent.